core/vm/contracts.go

```diff
@@ -202,30 +203,21 @@ func init() {
 	addrsList = append(addrsList, PrecompiledAddressesApricotPhasePre6...)
 	addrsList = append(addrsList, PrecompiledAddressesApricotPhase6...)
 	addrsList = append(addrsList, PrecompiledAddressesBanff...)
+	addrsList = append(addrsList, genesisContractAddr, NativeAssetBalanceAddr, NativeAssetCallAddr) // Add genesis multicoin contract and deprecated native asset precompiles to set of reserved addresses
 	for _, k := range addrsList {
 		PrecompileAllNativeAddresses[k] = struct{}{}
 	}
 
 	// Ensure that this package will panic during init if there is a conflict present with the declared
 	// precompile addresses.
-	for _, k := range precompile.UsedAddresses {
-		if _, ok := PrecompileAllNativeAddresses[k]; ok {
-			panic(fmt.Errorf("precompile address collides with existing native address: %s", k))
-		}
-		if k == constants.BlackholeAddr {
-			panic(fmt.Errorf("cannot use address %s for stateful precompile - overlaps with blackhole address", k))
+	for _, module := range modules.RegisteredModules() {
+		address := module.Address
+		if _, ok := PrecompileAllNativeAddresses[address]; ok {
+			panic(fmt.Errorf("precompile address collides with existing native address: %s", address))
 		}
 
-		// check that [k] belongs to at least one ReservedRange
-		found := false
-		for _, reservedRange := range precompile.ReservedRanges {
-			if reservedRange.Contains(k) {
-				found = true
-				break
-			}
-		}
-		if !found {
-			panic(fmt.Errorf("address %s used for stateful precompile but not specified in any reserved range", k))
+		if address == constants.BlackholeAddr {
```

Comment:
why do we check this here vs in subnet-evm where it's checked here https://github.com/ava-labs/subnet-evm/blob/32514d6c7c6523d08a07eadda9919ef2253232de/precompile/modules/registerer.go#L52-L53

Reply:
Removing